controllers/workspace_controller_variable_sets_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package controllers
+
+import (
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	tfc "github.com/hashicorp/go-tfe"
+	appv1alpha2 "github.com/hashicorp/hcp-terraform-operator/api/v1alpha2"
+)
+
+var _ = Describe("Workspace controller", Ordered, func() {
+	var (
+		instance       *appv1alpha2.Workspace
+		namespacedName types.NamespacedName
+		workspace      string
+
+		variableSetName  string
+		variableSetName2 string
+		variableSetID    string
+		variableSetID2   string
+	)
+
+	BeforeAll(func() {
+		// Set default Eventually timers
+		SetDefaultEventuallyTimeout(syncPeriod * 4)
+		SetDefaultEventuallyPollingInterval(2 * time.Second)
+	})
+
+	BeforeEach(func() {
+		namespacedName = newNamespacedName()
+		workspace = fmt.Sprintf("kubernetes-operator-%v", randomNumber())
+		variableSetName = fmt.Sprintf("variable-set-%v", randomNumber())
+		variableSetName2 = fmt.Sprintf("%v-2", variableSetName)
+		variableSetID = createTestVariableSet(variableSetName)
```

Comment:
Please validate this function. This set of tests fails due to an error returned by HCP Terraform.

Reply:
This is still an issue with the latest commit.